### PR TITLE
Handle more advanced Washing Machine features

### LINKED ISSUE #85: Support for settings
In the washing machine, oven and dishwasher configs there are settings defined. We should handle them

```
  "settings": [
    {
      "strKey": "SETTINGS_VOLUME",
      "wifiArrayIndex": 55,
      "enumValues": [
        {
          "strKey": "VOLUME_LOW",
          "wifiArrayValue": 0
        },
        {
          "strKey": "VOLUME_HIGH",
          "wifiArrayValue": 1
        }
      ]
    }
  ],
```

### LINKED ISSUE #86: Support for deviceWarnings
In all of the configs there are deviceWarnings defined. We should handle them

```
  "deviceWarnings": {
    "wifiArrayReadIndex": 59,
    "warnings": [
      {
        "strKey": "WASHER_WARNING_DOOR_IS_OPEN",
        "bitIndex": 0,
        "notificationInfo": {
          "strKey": "WASHER_WARNING_DOOR_IS_OPEN_NOTIFICATION",
          "priority": "HIGH",
          "necessity": ""
        }
      },
      {
        "strKey": "WASHER_WARNING_NO_WATER",
        "bitIndex": 1,
        "notificationInfo": {
          "strKey": "WASHER_WARNING_NO_WATER_NOTIFICATION",
          "priority": "HIGH",
          "necessity": ""
        }
      },
      {
        "strKey": "WASHER_WARNING_SECURITY",
        "bitIndex": 2,
        "notificationInfo": {
          "strKey": "WASHER_WARNING_SECURITY_NOTIFICATION",
          "priority": "BLOCKER",
          "necessity": ""
        }
      }
    ]
  },
```

### LINKED ISSUE #87: Support for remoteControl sensor/switch
In the washing machine, oven and dishwasher configs there is remoteControl section defined. We should handle it.

```
  "remoteControl": {
    "wifiArrayReadIndex": 32,
    "wifiArrayValue": 170
  },
```

### PR DESCRIPTION
**Pull request by rowysock**
Closes #85, 
Closes #86  
Closes #87

### CODE REVIEW

File: custom_components/homewhiz/appliance_controls.py [hunk 1]
```diff
@@ -307,6 +344,35 @@ def build_controls_from_progress_variables(
     return result
 
 
+def build_control_from_remote_control(remote_control: Optional[ApplianceRemoteControl]):
```
Comment (TechHummel) [inline custom_components/homewhiz/appliance_controls.py:347]: Maybe add return type since mypy is now part of pre-commit?

**Review by TechHummel — APPROVED** [review]

**Comment by TechHummel** [comment]
I tested the new functions in home assistant and it looks good. However, there is a new control element where I am not sure about its function
image. 
I'm planning on investigating this in the next couple of weeks.

Something else I noticed that is a bit strange is the sorting of entries:
image
image

I would think the order is copied from the files from the API, but I don't think I saw this order in the app 🤔

File: custom_components/homewhiz/appliance_controls.py (hunk 1 above)
Comment (rowysock) [reply custom_components/homewhiz/appliance_controls.py]: Done. added https://mypy.readthedocs.io/en/stable/config_file.html#confval-disallow_untyped_defs

**Comment by rowysock** [comment]
I've fixed the order

Custom duration level is just one of the controls from the homewhiz app, from the "advanced" section

**Pull request merged**